quodlibet/ext/events/advanced_preferences.py

```diff
@@ -114,6 +114,15 @@ def changed(entry, name, section="settings"):
                  "the ones present in the song list. Separate with \",\"")),
             text_config("settings", "rating_symbol_full", "Rating symbol (full):"),
             text_config("settings", "rating_symbol_blank", "Rating symbol (blank):"),
+            int_config(
+                "settings", "weight_smoothing_factor",
+                "Weight smoothing factor:",
+                ("This is used to smooth the length-based weights. Without smoothing, "
+                 "very long songs would dominate an average value of an album. "
+                 "At the same time, short good songs would have no affect on, for "
+                 "example, the average rating of an album. "
+                 "If this is -1 or less, averages are unweighted. "
+                 "If this is 0, no smoothing is applied. (Default is 10)")),
```

Comment:
```suggestion
                 "Set to 0 for no smoothing (...)")),
```
Though I'm not sure what weighted-but-smoothed means actually?

Reply:
Why the `(...)`?

> I'm not sure what weighted-but-smoothed means actually

The way I see it a `weight_smoothing_factor` of `0` implies that weights aren't smoothed. 

So in an album with two songs, where song 1 is twice as long as song 2 would result in: 

* song 1: has a weight of 2/3 (twice as much as song 2)
* song 2: has a weight of 1/3

(And, of course, then we calculate the rating like this for `:avg`: `s1.rating * s1.weight + s2.rating * s2.weight`)

Or did you mean something else?

Reply:
Right, I see, thanks :+1: 

Perhaps I'm hoping for a single config number where 
0 : no weighting
1: full (linear) time-weighting (like your example)

And assume nobody would want _more_ than full time-based weighting, that seems crazy.
Is this doable?

Reply:
Theoretically we could add two options: `use_duration_weights_in_average` and the existing one

---

Okay, GitHub seems buggy. For me, it's not displaying my last response at the bottom to at least one of your comment. I only see it via the ["Changed Files" tab](https://github.com/quodlibet/quodlibet/pull/3626/files), so you might have to check there for my response to the "Actually remind me why we're adding" comment you made.

Reply:
Right found it, thanks.

I think my overall suggestion is why not leave the _explicit_ aggregators (e.g. `avg`) unchanged, so that no user query results suddenly change, but change the _default aggregator_ for given tags.

So implementation-wise, change [`NUM_DEFAULT_FUNCS`](https://github.com/quodlibet/quodlibet/blob/2b7f645bf8120db60817992801f762ff5df340e7/quodlibet/util/collection.py) to point to the new weighted average (whatever it gets called) for whichver tags we choose (possibly all of these. 

This way nobody gets surprised, unless they were using defaults which will be documented to have changed, erm probably, and full choice is still available.

Another way is to ship closer to what you have here, but with the default smoothing to be turned off, to achieve the same "least surprise" outcome.

Either way we should think about ways of describing to users (or a tour of) "what's new" (see #1090 a bit, but much more than this), this would help lots of things

Reply:
@declension I still feel like that has the problem I described in the comment:

> At the same time, hiding good changes in the config means that many, many people will never use them.. For example, I've only now enabled bayesian averaging, despite using QL all these years, as I just didn't know why I should enable it. Other things were more important than figuring out what it is. I'm sure a lot of users are far less curious about why bayesian averaging, smoothing, or duration-weighting makes sense.

(which is also why I created #3634)

Here, if `rating:avg` stays the same, then most people will never benefit from the (small) change, as many people have already created most of their queries, or like me, they tend to use the `avg` even though it can be left out.

Of course, I get your point. There's a balance to be made between improving things for the better and some sort of backward compatibility / not surprising people. 

Hmm.. a poll would be useful in this case. (Is it okay to make the average for ratings length-weighted by default (both `~#rating:avg` and `~#rating`)? Yes, No, Just show results)